src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -2,49 +2,58 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using System.Runtime.InteropServices;
 using System.Threading;
 using System.Threading.Tasks;
-using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CSharp.RuntimeBinder;
+using Microsoft.VisualStudio.Editor;
 using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
 using Microsoft.VisualStudio.TextManager.Interop;
 using Roslyn.Utilities;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.Snippets
 {
+    /// <summary>
+    /// This service is created on the UI thread during package initialization, but it must not
+    /// block the initialization process. If the expansion manager is an IExpansionManager,
+    /// then we can use the asynchronous population mechanism it provides. Otherwise, getting
+    /// snippet information from the <see cref="IVsExpansionManager"/> must be done synchronously
+    /// through on the UI thread, which we do after package initialization at a lower priority.
+    /// </summary>
+    /// <remarks>
+    /// IExpansionManager was introduced in Visual Studio 2015 Update 1, but
+    /// will be enabled by default for the first time in Visual Studio 2015 Update 2. However,
+    /// the platform still supports returning the <see cref="IVsExpansionManager"/> if a major
```

Comment:
Sounds good.  As this point we don't have a solid date when the native code will be completely deprecated.